pytorch_pfn_extras/training/extensions/timeline_trace.py

```diff
@@ -0,0 +1,106 @@
+            formatting. For example, users can use '{iteration}' to separate
+            the log files for different iterations. If the log name is None, it
+            does not output the log to any file.
+        writer (writer object, optional): must be callable.
+            object to dump the log to. If specified, it needs to have a correct
+            `savefun` defined. The writer can override the save location in
+            the :class:`pytorch_pfn_extras.training.ExtensionsManager` object
+        tracer (tracer object, optional):
+            object with the tracing logic.  If not specified, the default
+            tracer in the thread local storage with be used.
+    Args:
+        entries (list): list of str
+
+    Returns:
+        header (str): header string
+        templates (str): template string for print values.
+    """
+
+    def __init__(
+        self,
+        trigger: trigger_module.TriggerLike = (1, "epoch"),
+        filename: Optional[str] = None,
+        enable: Optional[trigger_module.TriggerLike] = None,
+        disable: Optional[trigger_module.TriggerLike] = None,
+        **kwargs: Any,
+    ):
+        self._tracer = kwargs.get("tracer", get_tracer())
+        self._enable = None
+        if enable is not None:
+            self._enable = trigger_module.get_trigger(enable)
+
+        self._disable = None
+        if disable is not None:
+            self._disable = trigger_module.get_trigger(disable)
+
+        self._trigger = trigger_module.get_trigger(trigger)
+
+        self._filename = "chrome_trace.json" if filename is None else filename
+        self._writer = kwargs.get("writer", None)
+
+    def _flush_trace(self, manager: ExtensionsManagerProtocol) -> None:
+        writer = manager.writer if self._writer is None else self._writer
+
+        # write to the log file
+        if self._filename is not None:
+            self._tracer.flush(self._filename, writer)
+
+    def __call__(self, manager: ExtensionsManagerProtocol) -> None:
+        if self._enable is not None and self._enable(manager):
+            self._tracer.enable(True)
+        if self._disable is not None and self._disable(manager):
+            self._tracer.enable(False)
+
+        if not manager.is_before_training and self._trigger(manager):
+            self._flush_trace(manager)
+
+    def state_dict(self) -> Dict[str, Any]:
+        state: Dict[str, Any] = {}
+        if hasattr(self._trigger, "state_dict"):
+            state["_trigger"] = self._trigger.state_dict()
+        state["_tracer"] = self._tracer.state_dict()
+        return state
+
+    def load_state_dict(self, to_load: Dict[str, Any]) -> None:
+        if hasattr(self._trigger, "load_state_dict"):
+            self._trigger.load_state_dict(to_load["_trigger"])
+        self._tracer.load_state_dict(to_load["_tracer"])
+
+    def finalize(self, manager: ExtensionsManagerProtocol) -> None:
+        self._flush_trace(manager)
```

Comment:
I think that at the timing of finilize in extension, `synchronize()` in tracer should be executed.